### ISSUE TITLE
Create chain code for policies

### ISSUE THREAD

**Issue by AhmadBAmzah** [sardap/Capstone-2019-Data-Sharing#179]
Create chain code for our policies so only the wallet id can append to the active field.

**Comment by sardap** [comment]
Hi this is a summray of the deploy process to updating the active filed
1. Policy deployer genrates a random key and deploys all polices to one stream
2. Returns the trans ID and the key
3. later the following rpc call can be made for mulitchian to update the active field 
```
publish stream1 KEY '{"json":{"active" : [ true ]}}'
```
4. This will create a new tras in that stream for that key.
5. Once this trans is complete we can see that it has updated by getting the policy again example 
```
getstreamkeysummary stream1 KEY jsonobjectmerge,ignoreother
```
This means what we need the stream filter to do is to enfoce the following rules:
* Restrict publishing to that streams key to just the wallet ID (If this is not possible we will can move it so each policy has their own stream)

**Comment by AhmadBAmzah** [comment]
@sardap Is this what you wanted?
```javascript
function filterstreamitem()
{
	var item = getfilterstreamitem();
	if(item.keys[0] == "KEY" && item.publishers[0] != "WALLET_ID")
	{
		return "Only wallet id can append using key";
	}
}
```
Note: publishers refers to wallet address

**Comment by sardap** [comment]
Is there anyway to get an existing value from the depolyed policy eg get the wallet id from the deployed policy?

**Comment by AhmadBAmzah** [comment]
No, there is no way to access the previous stream item to get the deployed json policy. The filter only has access to the stream item being validated.

**Comment by sardap** [comment]
So we have no way of getting the wallet id?

**Comment by sardap** [comment]
Becuase this is the new plan. 
When policy is deployed a new stream is made with a smart filter with the wallet id hardcoded into the smart filter.

**Comment by AhmadBAmzah** [comment]
```javascript
function filterstreamitem()
{
	var item = getfilterstreamitem();
	var itemjson = item.data.json;
	if(itemjson.foo != "bar")
	{
		return "Item is rejected";
	}
}
```
Note:
- Example json stored on stream: ```{"foo": "bar"}```

**Comment by sardap** [comment]
```json
function filterstreamitem() 
{ 
    var item=getfilterstreamitem(); 
    if (item.publishers[0] != WALLET_ID ) 
        return "Only data subject can modify policy"; 
    if (item.keys.length > 1 || item.keys.indexOf("policy") != 0) 
        return "can only change the policy key"; 
    if (Object.keys(item.data.json).length != 1) 
        return "Can Only change the active field: "; 
    // This breaks if you make it not use the "true" and use true why No idea
    if(item.data.json.active != "true" && item.data.json.active != "false") 
        return "Can only set the active field to true or false" }
```
